core/chains/evm/client/pool.go

```diff
@@ -172,6 +182,32 @@ func (p *Pool) nLiveNodes() (nLiveNodes int, blockNumber int64, totalDifficulty
 	return
 }
 
+func (p *Pool) checkLease() {
+	bestNode := p.nodeSelector.Select()
+	for _, n := range p.nodes {
+		// Terminate client subscriptions. Services are responsible for reconnecting, which will be routed to the new
+		// best node. Only terminate connections with more than 1 subscription to account for the aliveLoop subscription
```

Comment:
Is there a condition in the node lifecycle that exits the alive loop if there are no **subscriptions**? Because if the RPC connection remains open, I have a feeling the alive loop won't be affected.

Reply:
Unfortunately, yes, terminating the alive loop subscription will lead to the node being declared unreachable. It will recover quickly, but it's undesirable behavior.

Reply:
Can you point me to the part of the code that does that so I can take a closer look?

Reply:
[Here](https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/client/node_lifecycle.go#L94) is the subscription from the alive loop and [here](https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/client/node_lifecycle.go#L182) is where the node will be declared unreachable when the subscription channel is closed